pkg/stanza/adapter/storage.go

```diff
@@ -25,37 +26,64 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
 )
 
-func GetStorageClient(ctx context.Context, id config.ComponentID, componentKind component.Kind, host component.Host) (storage.Client, error) {
-	var storageExtension storage.Extension
-	if host != nil {
-		for _, ext := range host.GetExtensions() {
-			if se, ok := ext.(storage.Extension); ok {
-				if storageExtension != nil {
-					return nil, errors.New("multiple storage extensions found")
-				}
-				storageExtension = se
-			}
+func GetStorageExtension(ctx context.Context, host component.Host, storageID config.ComponentID) (storage.Extension, error) {
+	if host == nil {
+		return nil, nil
+	}
+
+	// Storage explicitly disabled. (e.g. 'storage: false')
+	if storageID.String() == "false" {
```

Comment:
We probably need to add somewhere in the core a check that "false" is not a valid extension type.

Also, should this be "false" or "none" as the PR description suggests?